graphql_compiler/compiler/emit_sql.py

```diff
@@ -205,25 +289,66 @@ def _set_group_by(self, group_by):
         """Set output columns for the fold object."""
         self._group_by = group_by
 
-    def _construct_fold_joins(self, edge, from_alias, to_alias):
+    def _construct_fold_joins(self):
         """Use the edge descriptors to create the join clause between the tables in the fold."""
-        join_clause = sqlalchemy.join(
-            from_alias,
-            to_alias,
-            onclause=(from_alias.c[edge.from_column] == to_alias.c[edge.to_column])
-        )
+        # Start the join clause with the from_table of the first join descriptor
+        _, join_clause, _ = self.join_info[0]
+
+        # Starting at the first from_table, join traversed vertices until the output vertex is
+        # reached.
+        #
+        # MSSQL doesn't use the JOIN for the very last traversal,
+        # so we pop it off before construction the JOIN clause
+        if self._dialect == 'mssql':
+            # Move the final join predicate (from the vertex preceding the output to
+            # the output vertex) into the WHERE clause of the SELECT ... FOR XML PATH('')
+            self.join_info.pop()
+        while len(self.join_info) > 0:
+            # joins from earlier in the chain of traversals are at the end of the list
+            # b/c joins are appended in the order they are traversed
+            edge, from_alias, to_alias = self.join_info.pop(0)
+            join_clause = sqlalchemy.join(
+                join_clause,
+                to_alias,
+                onclause=(from_alias.c[edge.from_column] == to_alias.c[edge.to_column])
+            )
         return join_clause
 
     def _construct_fold_subquery(self, subquery_from_clause):
         """Combine all parts of the fold object to produce the complete fold subquery."""
-        return sqlalchemy.select(
+        select_statement = sqlalchemy.select(
             self.outputs
         ).select_from(
             subquery_from_clause
-        ).group_by(
-            *self.group_by
         )
 
+        if self._dialect == 'mssql':
+            # mssql doesn't rely on a group by
+            return select_statement
+        elif self._dialect == 'postgresql':
+            return select_statement.group_by(
+                *self.group_by
+            )
+        else:
+            raise NotImplementedError(
+                u'Fold only supported for MSSQL and '
+                u'PostgreSQL, dialect was set to {}'.format(self._dialect)
+            )
+
+    def _get_default_column(self, intermediate_fold_output_name, fold_output):
+        """Select an array_agg of the fold output field, labeled as requested."""
+        return sqlalchemy.func.array_agg(
+            self.output_vertex_alias.c[fold_output.field]
+        ).label(intermediate_fold_output_name)
+
+    def _get_mssql_column(self, intermediate_fold_output_name, fold_output, last_join):
```

Comment:
Please choose a better name here as well — and do a pass over the rest of the PR and consider if there are any other names you could improve.